src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/ExecutableStatementCountCheck.java

```diff
@@ -95,7 +95,6 @@ public void setMax(int max) {
     @Override
     public void beginTree(DetailAST rootAST) {
         context = new Context(null);
-        contextStack.clear();
```

Comment:
@Nimfadora , here is a nuance. 
All checks have to clear all internal and non-property fields in beginTree method to make sure that no context is shared between two files validation. For all files  we create only one instance of Check object, so on beginTree it has to reset all to initial state. 
To reproduce a problem, please launch Check on few files. 
Yes, we usually test Check on one file only in UT, but in real life one instance of Check is used for all files of execution.  
You need to make UT that executes Check on two files.